dlib/media/ffmpeg_muxer.h

```diff
@@ -1381,6 +1424,39 @@ namespace dlib
         inline AVCodecID        muxer::get_audio_codec_id()     const noexcept { return st.encoder_audio.get_codec_id(); }
         inline std::string      muxer::get_audio_codec_name()   const noexcept { return st.encoder_audio.get_codec_name(); }
 
+// ---------------------------------------------------------------------------------------------------
+
+        template <
+          class image_type,
+          is_image_check<image_type>
+        >
+        inline void save_frame(
+            const image_type& image,
+            const std::string& file_name,
+            const std::unordered_map<std::string, std::string>& codec_options
+        )
+        {
+            muxer writer([&] {
+                muxer::args args;
+                args.filepath               = file_name;
+                args.enable_image           = true;
+                args.enable_audio           = false;
+                args.args_image.h           = num_rows(image);
+                args.args_image.w           = num_columns(image);
+                args.args_image.framerate   = 1;
+                args.args_image.fmt         = pix_traits<pixel_type_t<image_type>>::fmt;
+                args.args_image.codec_options = codec_options;
+                args.format_options["update"] = "1";
+                return args;
+            }());
+
+            if (!writer.is_open())
+                throw error("ffmpeg::save_frame: error while saving " + file_name);
+
+            if (!writer.push(image))
+                throw error("ffmpeg::save_frame: error while saving " + file_name);
```

Comment:
Do you think we should do this? And similarly add `EIMAGE_LOAD` in the `ffmpeg::load_frame` function?
```suggestion
                throw error(EIMAGE_SAVE, "ffmpeg::save_frame: error while saving " + file_name);
```

Reply:
Yeah, why not. I'm not familiar with that, but if it's in keeping with the rest of dlib, then go for it.

Reply:
Let's see what @davisking thinks, I'd go for it, but I'm also not that familiar with that. It seems better than just having a generic error.

Reply:
yeah ok, feel free to commit

Reply:
sod it, done it.